src/main/java/com/redhat/cloud/notifications/routers/BehaviorGroupMigrationService.java

```diff
@@ -0,0 +1,317 @@
+                                                                                for (EndpointTarget target : targets) {
+                                                                                    mBehaviorGroup.actions.add(target.getEndpoint());
+                                                                                }
+                                                                                LOGGER.debugf("[BG MIGRATION] ______ %d action(s) added", targets.size());
+                                                                                aggregator.aggregate(mBehaviorGroup, Optional.of(eventType), report);
+                                                                                LOGGER.debugf("[BG MIGRATION] ______ 1 event type added to the links set", targets.size());
+                                                                            });
+                                                                });
+                                                    })
+                                                    .onItem().ignoreAsUni()
+                                                    .onItem().invoke(ignored -> LOGGER.debug("[BG MIGRATION] ____ Done"));
+                                        })
+                                        .onItem().ignoreAsUni()
+                                        .onItem().invoke(ignored -> LOGGER.debug("[BG MIGRATION] __ Done"))
+                                )
+                                .onItem().invoke(ignored -> {
+                                    LOGGER.debug("[BG MIGRATION] Done");
+                                    LOGGER.debug("[BG MIGRATION] Starting DB behavior groups creation");
+                                })
+                                /*
+                                 * Now we'll iterate over the aggregator entries.
+                                 * Each entry key is a behavior group and its value is a Set containing all event types that should be linked with the behavior group.
+                                 */
+                                .replaceWith(aggregator.entrySet())
+                                .onItem().transformToMulti(aggregatorEntries -> Multi.createFrom().iterable(aggregatorEntries))
+                                .onItem().transformToUniAndConcatenate(aggregatorEntry -> {
+                                    // It's time to finally persist a behavior group.
+                                    BehaviorGroup behaviorGroup = new BehaviorGroup();
+                                    behaviorGroup.setAccountId(aggregatorEntry.getKey().accountId);
+                                    behaviorGroup.setBundle(aggregatorEntry.getKey().bundle);
+                                    behaviorGroup.setBundleId(aggregatorEntry.getKey().bundle.getId()); // Yes, we need to set that even if the bundle was already set.
```

Comment:
Not to be done here - but wondering if in the future we could let the `setBundle` call to set the `bundleId` internally on these cases.

Reply:
This code is very unusual and temporary. Normally we would receive the `bundleId` from a REST request, retrieve the `Bundle` instance from the database and then call `setBundle`. So setting the `bundleId` from `setBundle` wouldn't really be useful.